### ISSUE TITLE
🛠 Tooling: Bump Knip to >=2.2.0 and remove ignoreDependencies: ["all-contributors-cli"]

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/template-typescript-node-package#334]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

[zx](https://github.com/google/zx) is a nifty tool for writing JS scripts that call to the shell. For example, this is a file that would run `ls`:

```js
await $`ls`;
```

Older versions of Knip didn't know that packages whose CLIs are run in this way are used. In #315, we had to add a Knip `ignoreDependencies` entry for `all-contributors-cli` only ever referenced in a script run by zx:

```js
await $`pnpm all-contributors generate`;
```

https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/02539981a28956acb6eda85cb9f63753e24aa416/knip.jsonc#LL4C3-L4C21

As of https://github.com/webpro/knip/issues/96, Knip >=2.2.0 should support detecting `zx` commands that run a package CLI. So we should be able to remove the `ignoreDependencies` entry from the `knip.jsonc`! Yay!

### Additional Info

Note that per https://github.com/webpro/knip/issues/96#issuecomment-1487585801, we'll need to add this on top of any file run by `zx`:

```js
#!/usr/bin/env zx
```

**Comment by navin-moorthy** [comment]
What's your thought on using `execa` [release v7.1.0](https://github.com/sindresorhus/execa/releases/tag/v7.1.0) with script interface ?

[`execa` vs `zx` comparsion](https://github.com/sindresorhus/execa/blob/main/docs/scripts.md).

**Comment by JoshuaKGoldberg** [comment]
@navin-moorthy thanks for bringing this up! I filed a new issue, #340 - in general it's good to post new issues for new topics instead of asking on unrelated issues.

**Comment by JoshuaKGoldberg** [comment]
Blocked on https://github.com/webpro/knip/issues/108 & https://github.com/webpro/knip/issues/109...

**Comment by navin-moorthy** [comment]
Right. I just din't think it as an issue rather a discussion.
